Unable to build ngtcp2 without wolfssl

Unable to build ngtcp2 without wolfssl support.

I am trying to build ngtcp2 but getting below error.

```
Makefile:4169: update target 'check-am' due to: all-am
make  examplestest
make[2]: Entering directory '/home/vagrant/quic_services/ngtcp2/examples'
Makefile:1469: update target 'examplestest' due to: examplestest-examplestest.o examplestest-util_test.o examplestest-util.o examplestest-siphash_test.o examplestest-siphash.o ../tests/munit/examplestest-munit.o ../lib/libngtcp2.la ../third-party/liburlparse.la
rm -f examplestest
echo "  CXXLD   " examplestest;/bin/bash ../libtool --silent --tag=CXX   --mode=link clang++-20 -std=gnu++20   -g -O2 -no-install  -Wl,-rpath,/home/vagrant/quic_services/ngtcp2/../openssl/build/lib -o examplestest examplestest-examplestest.o examplestest-util_test.o examplestest-util.o examplestest-siphash_test.o examplestest-siphash.o ../tests/munit/examplestest-munit.o  ../lib/libngtcp2.la ../third-party/liburlparse.la -lev -L/home/vagrant/quic_services/nghttp3/build/lib -lnghttp3
  CXXLD    examplestest
/usr/bin/ld: examplestest-examplestest.o: warning: relocation against `_ZN6ngtcp29sim_suiteE' in read-only section `.text.startup'
/usr/bin/ld: examplestest-examplestest.o: in function `main':
/home/vagrant/quic_services/ngtcp2/examples/examplestest.cc:43: undefined reference to `ngtcp2::sim_suite'
/usr/bin/ld: /home/vagrant/quic_services/ngtcp2/examples/examplestest.cc:43: undefined reference to `ngtcp2::sim_suite'
/usr/bin/ld: warning: creating DT_TEXTREL in a PIE
clang++-20: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [Makefile:1470: examplestest] Error 1
make[2]: Leaving directory '/home/vagrant/quic_services/ngtcp2/examples'
make[1]: *** [Makefile:4169: check-am] Error 2
make[1]: Leaving directory '/home/vagrant/quic_services/ngtcp2/examples'
make: *** [Makefile:568: check-recursive] Error 1
root@ubuntu:/home/vagrant/quic_services/ngtcp2#
root@ubuntu:/home/vagrant/quic_services/ngtcp2#
root@ubuntu:/home/vagrant/quic_services/ngtcp2#
root@ubuntu:/home/vagrant/quic_services/ngtcp2#
root@ubuntu:/home/vagrant/quic_services/ngtcp2#
root@ubuntu:/home/vagrant/quic_services/ngtcp2# git log
commit 67eda3001f9f11466cada0c7aa037bb482e07c42 (HEAD -> main, origin/main, origin/HEAD)
Merge: ba1ad342 596d8b2c
Author: Tatsuhiro Tsujikawa <404610+tatsuhiro-t@users.noreply.github.com>
Date:   Mon Nov 17 23:18:44 2025 +0900

    Merge pull request #1907 from ngtcp2/bump-libressl

    Bump libressl to v4.2.1

commit 596d8b2c8203274f3118299b91c335e1747c5964
Author: Tatsuhiro Tsujikawa <tatsuhiro.t@gmail.com>
Date:   Mon Nov 17 22:02:36 2025 +0900

    Bump libressl to v4.2.1

commit ba1ad342195370da233908956bc83186a70f03b9
Merge: 5bc53efd 0b56f759
Author: Tatsuhiro Tsujikawa <404610+tatsuhiro-t@users.noreply.github.com>
Date:   Mon Nov 17 21:39:41 2025 +0900

    Merge pull request #1905 from ngtcp2/sim

    examples: Add simulator
```

Am I missing anything?

--- a/examples/Makefile.am
+++ b/examples/Makefile.am
@@ -273,7 +273,7 @@ examplestest_SOURCES += \
 	sim.cc sim.h \
 	sim_test.cc sim_test.h \
 	debug.cc debug.h
-examplestest_CPPFLAGS += @WOLFSSL_CFLAGS@
+examplestest_CPPFLAGS += @WOLFSSL_CFLAGS@ -DWITH_EXAMPLE_WOLFSSL
 examplestest_LDADD += \
 	$(top_builddir)/crypto/wolfssl/libngtcp2_crypto_wolfssl.la \
 	@WOLFSSL_LIBS@

--- a/examples/examplestest.cc
+++ b/examples/examplestest.cc
@@ -32,13 +32,17 @@
 // include test cases' include files here
 #include "util_test.h"
 #include "siphash_test.h"
-#include "sim_test.h"
+#ifdef WITH_EXAMPLE_WOLFSSL
+#  include "sim_test.h"
+#endif // defined(WITH_EXAMPLE_WOLFSSL)
 
 int main(int argc, char *argv[]) {
   const MunitSuite suites[] = {
     ngtcp2::util_suite,
     ngtcp2::siphash_suite,
+#ifdef WITH_EXAMPLE_WOLFSSL
     ngtcp2::sim_suite,
+#endif // defined(WITH_EXAMPLE_WOLFSSL)
     {},
   };
   const MunitSuite suite = {

